pypfb/exporters.py

```diff
@@ -0,0 +1,120 @@
+import csv
+import gzip
+import os
+from uuid import uuid4
+
+from fastavro import reader
+
+from .avro_utils.avro_types import is_enum
+from .utils.str import decode
+
+
+class PFBExporter(object):
+    def __init__(self, file_obj):
+        self.reader = reader(file_obj)
+        self.fields_by_name = {}
+
+    def __enter__(self):
+        for field in self.reader.writer_schema["fields"]:
+            if field["name"] == "object":
+                it = iter(field["type"])
+                # skip metadata schema
+                next(it)
+                for t in it:
+                    self.fields_by_name[t["name"]] = t["fields"]
+        return self
+
+
+class PFB2GremlinExporter(PFBExporter):
+    type_mapping = dict(
+        boolean="Boolean",
+        enum="String",
+        integer="Long",
+        long="Long",
+        md5sum="String",
+        float="Double",
+        number="Double",
+        string="String",
+    )
+
+    def __init__(self, file_obj):
+        super(PFB2GremlinExporter, self).__init__(file_obj)
+        self._handlers_by_name = {}
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        for f, w in self._handlers_by_name.itervalues():
+            f.close()
+
+    def export_files(self, dir_path, gzipped=False):
+        uuids = {}
+        project_ids = []
+        edges = []
+
+        if not os.path.exists(dir_path):
+            os.mkdir(dir_path)
+
+        path = os.path.join(dir_path, "gremlin_edges.csv")
+        if gzipped:
+            f = gzip.open(path + ".gz", "wb")
+        else:
+            f = open(path, "wb")
+        ew = csv.writer(f)
+        ew.writerow(["~id", "~from", "~to", "~label"])
+        self._handlers_by_name["~edges"] = f, ew
+
+        it = iter(self.reader)
+        # skip metadata
+        next(it)
+        for row in it:
+            name = row["name"]
+            fields = self.fields_by_name[name]
+
+            row_id = row["id"]
+            obj = row["object"]
+            relations = row["relations"]
+
+            pair = self._handlers_by_name.get(name)
+            if pair is None:
+                header_row = ["~label", "~id"]
+                for field in fields:
+                    field_type = "string"
+                    for field_type in field["type"]:
+                        if field_type != "null":
+                            break
+                    if isinstance(field_type, (str, unicode)):
+                        field_type = self.type_mapping[field_type]
+                    else:
+                        field_type = "String"
```

Comment:
should you log some warning/info here about defaulting to "String"?

Reply:
added a comment only, because it is for complex types like `enum`